File: talkadvisor-domain/src/main/java/org/craftsrecords/talkadvisor/recommendation/spi/SearchTalks.java
package org.craftsrecords.talkadvisor.recommendation.spi;

import org.craftsrecords.talkadvisor.recommendation.preferences.Topic;
import org.craftsrecords.talkadvisor.recommendation.talk.Talk;

import javax.annotation.Nonnull;
import java.util.Set;

@FunctionalInterface
public interface SearchTalks {
    @Nonnull
    Set<Talk> forTopic(@Nonnull Topic topic);
}

File: talkadvisor-domain/src/main/kotlin/org/craftsrecords/talkadvisor/recommendation/spi/SearchTalks.kt
package org.craftsrecords.talkadvisor.recommendation.spi

import org.craftsrecords.talkadvisor.recommendation.preferences.Topic
import org.craftsrecords.talkadvisor.recommendation.talk.Talk

@FunctionalInterface
interface SearchTalks {
    val maxNumberOfTalks: Int
    fun forTopics(topics: Set<Topic>): Set<Talk>
}
